packages/react-storage/src/components/StorageBrowser/actions/handlers/createFolder.ts

```diff
@@ -24,5 +28,37 @@ export interface CreateFolderHandlerOutput extends TaskHandlerOutput {}
 export interface CreateFolderHandler
   extends TaskHandler<CreateFolderHandlerInput, CreateFolderHandlerOutput> {}
 
-export const createFolderHandler: CreateFolderHandler =
-  null as unknown as CreateFolderHandler;
+export const createFolderHandler: CreateFolderHandler = (input) => {
+  const { destinationPrefix, config, data, options } = input;
+  const { accountId, credentials, customEndpoint } = config;
+  const { onProgress, preventOverwrite } = options ?? {};
+  const { key } = data;
+
+  const bucket = constructBucket(config);
+
+  const { result } = uploadData({
+    path: `${destinationPrefix}${key}`,
+    data: '',
+    options: {
+      bucket,
+      expectedBucketOwner: accountId,
+      locationCredentialsProvider: credentials,
+      customEndpoint,
+      onProgress: (event) => {
+        if (isFunction(onProgress)) onProgress(data, getProgress(event));
+      },
+      preventOverwrite,
+    },
+  });
+
+  return {
+    result: result
+      .then(() => ({ status: 'COMPLETE' as const }))
+      .catch(({ message, name }: Error) => {
+        if (name === 'PreconditionFailed') {
+          return { status: 'OVERWRITE_PREVENTED' } as const;
```

Comment:
nit (good to do later as well):
we could move these constants to a separate file